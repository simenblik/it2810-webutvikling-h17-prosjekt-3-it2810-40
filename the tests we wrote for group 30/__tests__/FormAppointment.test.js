import React from 'react';
import { shallow, mount, render } from 'enzyme';
import toJson from 'enzyme-to-json';
import FormAppointment from '../src/Components/FormAppointment.js';

/* This is getting a warning. Need something to fix error:
console.error node_modules/fbjs/lib/warning.js:33
      Warning: Failed prop type: Invalid prop `children` of type `array` supplied to `MuiThemeProvider`, expected a single ReactElement.
          in MuiThemeProvider
*/

/* Takes a snapshot of the rendered component and see if it
matches the original snapshot.
*/
describe('<FormAppointment/>, functions', () => {
	it('should match to previous snapshots', () => {
		const component = shallow(<FormAppointment />);
		const tree = toJson(component);
		expect(tree).toMatchSnapshot();
	});
});
