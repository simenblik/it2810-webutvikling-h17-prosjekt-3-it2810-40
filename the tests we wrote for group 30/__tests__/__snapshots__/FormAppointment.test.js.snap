// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FormAppointment/>, functions should match to previous snapshots 1`] = `
<div
  id="appointmentContainer"
>
  <form
    onSubmit={[Function]}
  >
    <MuiThemeProvider>
      <DatePicker
        autoOk={false}
        container="dialog"
        disableYearSelection={false}
        disabled={false}
        firstDayOfWeek={0}
        hideCalendarDate={false}
        hintText="Dato"
        minDate={2017-10-18T20:52:24.012Z}
        onChange={[Function]}
        openToYearSelection={false}
        required={true}
        style={Object {}}
        value={null}
      />
    </MuiThemeProvider>
    <MuiThemeProvider>
      <TimePicker
        autoOk={false}
        cancelLabel="Cancel"
        defaultTime={null}
        disabled={false}
        format="24hr"
        hintText="Klokkeslett"
        minDate={2017-10-18T20:52:24.012Z}
        minutesStep={1}
        okLabel="OK"
        onChange={[Function]}
        pedantic={false}
        required={true}
        style={Object {}}
        value={null}
      />
    </MuiThemeProvider>
    <label>
      <textarea
        id="textArea"
        maxLength="50"
        onChange={[Function]}
        required={true}
      />
    </label>
    <input
      id="submitButton"
      type="submit"
      value="Add event"
    />
  </form>
  <ul
    id="appointmentlist"
  >
    <li
      id="appointmentitem"
      key="0"
    >
      <button
        id="removeAppointment"
        onClick={[Function]}
      >
        null
      </button>
    </li>
  </ul>
</div>
`;
